hadoop-ozone/client/src/main/java/org/apache/hadoop/ozone/client/OzoneKey.java

```diff
@@ -76,6 +80,22 @@ public OzoneKey(String volumeName, String bucketName,
             ReplicationFactor.valueOf(replicationFactor));
   }
 
+  @SuppressWarnings("parameternumber")
+  public OzoneKey(String volumeName, String bucketName,
+                  String keyName, long size, long creationTime,
+                  long modificationTime, ReplicationType type,
+                  int replicationFactor, Map<String, String> metadata) {
+    this.volumeName = volumeName;
+    this.bucketName = bucketName;
+    this.name = keyName;
+    this.dataSize = size;
+    this.creationTime = Instant.ofEpochMilli(creationTime);
```

Comment:
This should be set by OM?

Reply:
Also, nesting the constructors would help reduce code.

Reply:
looking into it